src/server/pfs/s3/object.go

```diff
@@ -43,14 +42,36 @@ func (c *controller) GetObject(r *http.Request, bucketName, file, version string
 		commitID = version
 	}
 
-	fileInfo, err := pc.InspectFile(bucket.Commit, file)
+	// We use listFileResult[0] rather than InspectFile result since InspectFile
+	// on a path that has both a file and a directory in it returns the
+	// directory. However, ListFile will show it as a file, if it exists.
+	var firstFile *pfs.FileInfo
+	err = pc.ListFile(bucket.Commit, file, func(fi *pfs.FileInfo) (retErr error) {
+		if firstFile == nil {
```

Comment:
So just to clarify my understanding, lets say we are in an open commit, and we have a situation like

```
NAME TYPE SIZE 
/0   file 1B   
/0/  dir  4B 
/0/foo file
```

The problem this is trying to solve is when Spark is trying to GetObject on `/0` and we want to return the `/0   file` instead of the dir.

Tho I'm a bit worried about the `/0/foo`, is it always guaranteed that the ListFile iteration always iterats on the `/0` first?

Reply:
That's right. In practice I've always seen the `/0` show up first.